app/services/invoices/payments/adyen_service.rb

```diff
@@ -0,0 +1,178 @@
+# frozen_string_literal: true
+
+module Invoices
+  module Payments
+    class AdyenService < BaseService
+      PENDING_STATUSES = %w[AuthorisedPending Received].freeze
+      SUCCESS_STATUSES = %w[Authorised SentForSettle SettleScheduled Settled Refunded].freeze
+      FAILED_STATUSES = %w[Cancelled CaptureFailed Error Expired Refused].freeze
+
+      def initialize(invoice = nil)
+        @invoice = invoice
+
+        super(nil)
+      end
+
+      def create
+        result.invoice = invoice
+        return result unless should_process_payment?
+
+        unless invoice.total_amount_cents.positive?
+          update_invoice_payment_status(payment_status: :succeeded)
+          return result
+        end
+
+        increment_payment_attempts
+
+        adyen_result = create_adyen_payment
+
+        payment = Payment.new(
+          invoice:,
+          payment_provider_id: adyen_payment_provider.id,
+          payment_provider_customer_id: customer.adyen_customer.id,
+          amount_cents: adyen_result.dig("amount", "value"),
+          amount_currency: adyen_result.dig("amount", "currency"),
+          provider_payment_id: adyen_result["pspReference"],
+          status: adyen_result["resultCode"],
+        )
+        payment.save!
+
+        invoice_payment_status = invoice_payment_status(payment.status)
+        update_invoice_payment_status(payment_status: invoice_payment_status)
+
+        result.payment = payment
+        result
+      end
+
+      def update_payment_status(provider_payment_id:, status:)
+        payment = Payment.find_by(provider_payment_id:)
+        return result.not_found_failure!(resource: 'adyen_payment') unless payment
+
+        result.payment = payment
+        result.invoice = payment.invoice
+        return result if payment.invoice.succeeded?
+
+        payment.update!(status:)
+
+        invoice_payment_status = invoice_payment_status(status)
+        update_invoice_payment_status(payment_status: invoice_payment_status)
+
+        result
+      rescue BaseService::FailedResult => e
+        result.fail_with_error!(e)
+      end
+
+      private
+
+      attr_accessor :invoice
+
+      delegate :organization, :customer, to: :invoice
+
+      def should_process_payment?
+        return false if invoice.succeeded?
+        return false if adyen_payment_provider.blank?
+
+        customer&.adyen_customer&.provider_customer_id
+      end
+
+      def client
+        @client ||= Adyen::Client.new(
+          api_key: adyen_payment_provider.api_key,
+          env: adyen_payment_provider.environment,
+          live_url_prefix: adyen_payment_provider.live_prefix
```

Comment:
Style/TrailingCommaInArguments: Put a comma after the last parameter of a multiline method call. (https://rubystyle.guide#no-trailing-params-comma)